File: BMI/app/src/main/java/com/example/bmi/CalculatorActivity.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class CalculatorActivity extends AppCompatActivity {

    private TextView bmiCounter, bmiGroup, bmiError;
    Integer kg, cm;
    private EditText weight, height;
    Double calculatedCount;
    String calculatedGroup;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculator);

        weight = (EditText) findViewById(R.id.editTextWeight);
        height = (EditText) findViewById(R.id.editTextHeight);

        bmiCounter = (TextView) findViewById(R.id.textViewBmiCount);
        bmiGroup = (TextView) findViewById(R.id.textViewBmiGroup);
        bmiError = (TextView) findViewById(R.id.textError);

        Button buttonCalculate = (Button) findViewById(R.id.buttonCalculate);
        Button buttonExit = (Button) findViewById(R.id.buttonExit);
        Button buttonKcal = (Button) findViewById(R.id.buttonKcal);
        Button buttonDish = (Button) findViewById(R.id.buttonDish);

        buttonCalculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try{
                    bmiError.setText("");
                    kg = Integer.parseInt(weight.getText().toString());
                    cm = Integer.parseInt(height.getText().toString());
                    calculatedCount = calculateCount(kg, cm);
                    calculatedGroup = calculateGroup(calculatedCount);

                    bmiCounter.setText(String.format("%.2f", calculatedCount));
                    bmiGroup.setText(String.valueOf(calculatedGroup));
                }catch (NumberFormatException err){
                    bmiError.setText(getString(R.string.text_error));
                }
            }
        });

        buttonKcal.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try{
                    bmiError.setText("");
                    if (calculatedGroup.equals("underweight") || calculatedGroup.equals("optimum") || calculatedGroup.equals("overweight")) {
                        launchShowKcal();
                    }
                }catch (NullPointerException err){
                    bmiError.setText(getString(R.string.text_error));
                }
            }
        });

        buttonDish.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try{
                    bmiError.setText("");
                    if (calculatedGroup.equals("underweight") || calculatedGroup.equals("optimum") || calculatedGroup.equals("overweight")) {
                        launchShowDish();
                    }
                }catch (NullPointerException err){
                    bmiError.setText(getString(R.string.text_error));
                }
            }
        });

        buttonExit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    protected Double calculateCount(Integer kg, Integer cm) {
        return kg / (Math.pow(cm,2)) * 10000;
    }

    protected String calculateGroup(Double calculatedCount) {
        if (calculatedCount <= 18.49) { return "underweight"; }
        else if (calculatedCount > 18.49 && calculatedCount <= 24.49) { return "optimum"; }
        else if (calculatedCount > 24.49 && calculatedCount <= 29.99) { return "overweight"; }
        else { return "obesity"; }
    }

    private void launchShowKcal() {
        Intent intent = new Intent(CalculatorActivity.this, ShowCaloriesActivity.class);
        intent.putExtra("KG", kg);
        intent.putExtra("CM", cm);
        startActivity(intent);
    }

    private void launchShowDish() {
        Intent intent = new Intent(this, ShowDishActivity.class);
        intent.putExtra("GROUP", calculatedGroup);
        startActivity(intent);
    }
}

File: BMI_kotlin/app/src/main/java/com/example/bmi_kotlin/CalculatorActivity.kt
package com.example.bmi_kotlin

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_calculator.*

class CalculatorActivity : AppCompatActivity() {

    var kg: Int? = null
    var cm: Int? = null
    var calculatedCount: Double = 0.0
    var calculatedGroup: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculator)

        buttonCalculate.setOnClickListener {
            textError.text = ""

            if (editTextWeight.text.toString().length > 0
                    && editTextHeight.text.toString().length > 0
                    && !editTextWeight.text.toString().equals("0")
                    && !editTextHeight.text.toString().equals("0")) {
                kg = editTextWeight.text.toString().toInt()
                cm = editTextHeight.text.toString().toInt()

                calculatedCount = calculateCount(kg!!, cm!!)
                calculatedGroup = calculateGroup(calculatedCount!!)

                textViewBmiCount.text = calculatedCount.toString()
                textViewBmiGroup.text = calculatedGroup.toString()
            } else textError.text = getString(R.string.text_error)
        }

        buttonKcal.setOnClickListener {
            textError.text = ""

            if (editTextWeight.text.toString().length > 0
                    && editTextHeight.text.toString().length > 0
                    && !editTextWeight.text.toString().equals("0")
                    && !editTextHeight.text.toString().equals("0")) {
                val intent = Intent(this, ShowCaloriesActivity::class.java)
                intent.putExtra("KG", editTextWeight.text.toString().toInt())
                intent.putExtra("CM", editTextHeight.text.toString().toInt())
                startActivity(intent)
            } else textError.text = getString(R.string.text_error)
        }

        buttonDish.setOnClickListener {
            textError.text = ""

            if(calculatedGroup.equals("underweight")
                    || calculatedGroup.equals("optimum")
                    || calculatedGroup.equals("overweight")) {
                val intent = Intent(this, ShowDishActivity::class.java)
                intent.putExtra("GROUP", calculatedGroup)
                startActivity(intent)
            } else textError.text = getString(R.string.text_error)
        }
    }

    fun calculateCount(kg: Int, cm: Int) : Double {
        return kg / (Math.pow(cm.toDouble(), 2.0)) * 10000
    }

    fun calculateGroup(calculatedCount: Double) : String {
        if (calculatedCount <= 18.49) return "underweight"
        else if (calculatedCount > 18.49 && calculatedCount <= 24.49) return "optimum"
        else if (calculatedCount > 24.49 && calculatedCount <= 29.99) return "overweight"
        else return "obesity"
    }
}
